tests/components/fluss/__init__.py

```diff
@@ -0,0 +1,90 @@
+"""Test Script for Fluss+ Initialisation."""
+
+from __future__ import annotations
+
+from unittest.mock import MagicMock, patch
+
+from fluss_api import (
+    FlussApiClient,
+    FlussApiClientAuthenticationError,
+    FlussApiClientCommunicationError,
+    FlussApiClientError,
+)
+import pytest
+
+from homeassistant.components.fluss import PLATFORMS
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+
+@pytest.mark.parametrize(
+    "side_effect, expected_exception",
+    [
+        (FlussApiClientAuthenticationError, ConfigEntryAuthFailed),
+        (FlussApiClientCommunicationError, ConfigEntryNotReady),
+        (FlussApiClientError, ConfigEntryNotReady),
+    ],
+)
+async def test_async_setup_entry_errors(
+    hass: HomeAssistant,
+    mock_config_entry: MagicMock,
+    side_effect: Exception,
+    expected_exception: type[Exception],
+) -> None:
+    """Test setup errors."""
+    with (
+        patch("fluss_api.FlussApiClient", side_effect=side_effect),
+        pytest.raises(expected_exception),
+    ):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
```

Comment:
```suggestion
    with patch("homeassistant.components.fluss.coordinator.FlussApiClient", side_effect=side_effect):
        await hass.config_entries.async_setup(mock_config_entry.entry_id)
```
Just checking the behavior of the entry state is enough